### ISSUE TITLE
Can't create cluster with Azure CNI Overlay and NetworkPolicy

### ISSUE THREAD

**Issue by felfa01** [kubernetes-sigs/cluster-api-provider-azure#4093]
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
#3678 introduced support for [Azure CNI Overlay](https://learn.microsoft.com/en-us/azure/aks/azure-cni-overlay) on AKS managed clusters. However, there appears to be a bug when you create an AKS cluster with both `networkPolicy` and `networkPluginMode` enabled.
```yaml
networkPolicy: calico
networkPluginMode: overlay
```
With this configuration, the cluster creation does not happen and the capi-controller throws this error:
```
Reconciler error" err="[failed to retrieve Spec.ControlPlaneEndpoint from infrastructure provider for Cluster \"XXXXXXXX\" in namespace \"XXXXXXXXX\": field not found, admission webhook \"validation.azuremanagedcontrolplanes.infrastructure.cluster.x-k8s.io\" denied the request: AzureManagedControlPlane.infrastructure.cluster.x-k8s.io \"XXXXXXXXX\" is invalid: Spec.NetworkPluginMode: Forbidden: \"overlay\" NetworkPolicyMode cannot be enabled when NetworkPolicy is set]" controller="cluster" controllerGroup="cluster.x-k8s.io" controllerKind="Cluster" Cluster="XXXXXXX/XXXXXXX" namespace="XXXXXXX" name="XXXXXXX" reconcileID=b88e32a7-ea71-4769-a69f-ceb0196145e0
```
**What did you expect to happen:**
As can be seen in #3678 I actually tested this exact scenario with success at first. However, I did not do any follow up tests after the final commits happened on the PR after [this discussion.](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3988#discussion_r1323342321) This makes me think that the validation that was added introduces the bug, thus not fulfilling the criteria "You can create a cluster w/ CNI Overlay + NetworkPolicy when you create your cluster".

**Anything else you would like to add:**
```yaml
#networkPolicy: calico
networkPluginMode: overlay
```
Commenting out `networkPolicy` works fine.

**Environment:**

- cluster-api-provider-azure version: 1.11.2
- Kubernetes version: (use `kubectl version`): 1.26.6

**Comment by felfa01** [comment]
FYI @nojnhuh, I think we have a bug.

**Comment by nojnhuh** [comment]
@felfa01 You're right, I see a bug in the validation. PR open to fix: #4097
